odl/solvers/functional/default_functionals.py

```diff
@@ -2292,6 +2294,216 @@ def gradient(self):
                 (1 / self.sigma) * self.functional.proximal(self.sigma))
 
 
+class Huber(Functional):
+    """The Huber norm functional.
+
+    Notes
+    -----
+    The Huber norm is the integral over a smoothed norm. In detail, it is given
+    by
+
+    .. math::
+        F(x) = \\int_\Omega f_{\\gamma}(||x(y)||_2) dy
+
+    where ``||.||_2`` denotes the Euclidean norm for vector-valued functions
+    which reduces to the absolute value for scalar-valued functions.
+    The function :math:`f` with smoothing :math:`\\gamma` is given by
+
+    .. math::
+        f_{\\gamma}(t) =
+        \\begin{cases}
+            \\frac{1}{2 \\gamma} t^2 & \\text{if } |t| \leq \\gamma \\\\
+            |t| - \\frac{\\gamma}{2} & \\text{else}
+        \\end{cases}.
+    """
+
+    def __init__(self, space, gamma):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `FnBase`
+            Domain of the functional.
+        gamma : float
+            Smoothing parameter of Huberization. If ``gamma = 0``, then
+            functional is non-smooth corresponds to the usual L1 norm. For
+            ``gamma > 0``, it has a ``1/gamma``-Lipschitz gradient so that
+            its convex conjugate is ``gamma``-strongly convex.
+
+        Examples
+        --------
+        Example of initializing the Huber norm functional.
+
+        >>> space = odl.uniform_discr(0, 1, 14)
+        >>> gamma = 0.1
+        >>> huber_norm = odl.solvers.Huber(space, gamma=0.1)
+
+        Check that if all elements are > ``gamma`` we get the L1-norm up to a
+        constant.
+
+        >>> x = 2 * gamma * space.one()
+        >>> tol = 1e-5
+        >>> constant = gamma / 2 * space.one().inner(space.one())
+        >>> f = odl.solvers.L1Norm(space) - constant
+        >>> abs(huber_norm(x) - f(x)) < tol
+        True
+
+        Check that if all elements are < ``gamma`` we get the squared L2-norm
+        times the weight 1/(2*gamma).
+
+        >>> x = gamma / 2 * space.one()
+        >>> f = 1 / (2 * gamma) * odl.solvers.L2NormSquared(space)
+        >>> abs(huber_norm(x) - f(x)) < tol
+        True
+
+        Compare Huber- and L1-norm for vanishing smoothing ``gamma=0``.
+
+        >>> x = odl.phantom.white_noise(space)
+        >>> huber_norm = odl.solvers.Huber(space, gamma=0)
+        >>> l1_norm = odl.solvers.L1Norm(space)
+        >>> abs(huber_norm(x) - l1_norm(x)) < tol
+        True
+
+        Redo previous example for a product space in two dimensions.
```

Comment:
Colons at the end of those intermediate texts.

Reply:
done